### PR TITLE
Rename Burn to Slash in Assets

### PR DESCRIPTION
**Pull request by joepetrowski**
Renames `burn` to `slash` to update API for Statemint. Also removes duplicate `Burned` and `Issued` events.

### CODE REVIEW

File: frame/assets/src/lib.rs [hunk 1]
```diff
@@ -564,8 +564,7 @@ pub mod pallet {
 			let who = T::Lookup::lookup(who)?;
 
 			let f = DebitFlags { keep_alive: false, best_effort: true };
-			let burned = Self::do_burn(id, &who, amount, Some(origin), f)?;
-			Self::deposit_event(Event::Burned(id, who, burned));
```
Comment (joepetrowski) [inline frame/assets/src/lib.rs:568]: `do_burn` emits this event, shouldn't need it twice:
https://github.com/paritytech/substrate/blob/66c85aeb29fbf4786e76be328cf73241f72c2f18/frame/assets/src/functions.rs#L345

**Comment by apopiak** [comment]
Should we also adjust the event to `Slashed`?

**Comment by joepetrowski** [comment]
> Should we also adjust the event to `Slashed`?

I think it makes sense in the context of `assets`, but don't want to rename the helper functions (e.g. `do_burn`) as it creeps into the `fungibles` trait.

**Comment by Swader** [comment]
What is the rationale behind this change?

**Comment by apopiak** [comment]
@gavofyork Can you give your reasoning?

**Review by shawntabrizi — APPROVED** [review]

**Comment by Swader** [comment]
Still interested in learning more about the reasoning here. To me, slash has very clear connotations of "owner of asset having promised to do something and failing to deliver on that promise, thereby getting slashed". To me at least, slash clearly communicates punishment. Burn on the other hand covers both self-inflicted ops and burns from an admin, with or without a previous commitment (e.g. tether is ordered to burn some amount in my account, but also I can burn some amount of USDT in my account on my own - in neither case did I have a commitment to the network the dishonoring of which would punish me through a _slash_).

**Pull request closed without merging**